test/Microsoft.AspNetCore.Razor.Evolution.Test/Intermediate/DefaultRazorIRLoweringPhaseIntegrationTest.cs

```diff
@@ -188,9 +191,106 @@ public void Lower_WithUsing()
+                }
+            };
+
+            // Act
+            var irDocument = Lower(codeDocument, descriptors);
+
+            // Assert
+            Children(irDocument,
+                n => Assert.IsType<ChecksumIRNode>(n),
+                n => Assert.IsType<NamespaceDeclarationIRNode>(n));
+            var @namespace = irDocument.Children[1];
+            Children(@namespace,
+                n => Using("System", n),
+                n => Using(typeof(Task).Namespace, n),
+                n => Assert.IsType<ClassDeclarationIRNode>(n));
+            var @class = @namespace.Children[2];
+            Children(@class,
+                n => TagHelperFieldDeclaration(n, "SpanTagHelper"),
+                n => Assert.IsType<RazorMethodDeclarationIRNode>(n));
+            var method = @class.Children[1];
+            var tagHelperNode = SingleChild<TagHelperIRNode>(method);
+            Children(tagHelperNode,
+                n => TagHelperStructure("span", TagMode.StartTagAndEndTag, n),
+                n => Assert.IsType<CreateTagHelperIRNode>(n),
+                n => TagHelperHtmlAttribute(
+                    "val",
+                    HtmlAttributeValueStyle.DoubleQuotes,
+                    n,
+                    v => CSharpAttributeValue(string.Empty, "Hello", v),
+                    v => LiteralAttributeValue(" ", "World", v)),
+                n => Assert.IsType<ExecuteTagHelpersIRNode>(n));
+        }
+
+        [Fact]
+        public void Lower_TagHelpersWithBoundAttribute()
+        {
+            // Arrange
+            var codeDocument = TestRazorCodeDocument.Create(@"<input bound='foo' />");
+            var descriptor = new TagHelperDescriptor
+            {
+                TagName = "input",
+                TypeName = "InputTagHelper",
+                Attributes = new[] { new TagHelperAttributeDescriptor
+                {
+                    Name = "bound",
+                    PropertyName = "FooProp",
+                    TypeName = "System.String"
+                } }
+            };
+
+            // Act
+            var irDocument = Lower(codeDocument, new[] { descriptor });
+
+            // Assert
+            Children(irDocument,
+                n => Assert.IsType<ChecksumIRNode>(n),
+                n => Assert.IsType<NamespaceDeclarationIRNode>(n));
+            var @namespace = irDocument.Children[1];
+            Children(@namespace,
+                n => Using("System", n),
+                n => Using(typeof(Task).Namespace, n),
+                n => Assert.IsType<ClassDeclarationIRNode>(n));
+            var @class = @namespace.Children[2];
+            Children(@class,
+                n => TagHelperFieldDeclaration(n, "InputTagHelper"),
+                n => Assert.IsType<RazorMethodDeclarationIRNode>(n));
+            var method = @class.Children[1];
+            var tagHelperNode = SingleChild<TagHelperIRNode>(method);
+            Children(tagHelperNode,
```

Comment:
Could probably just make a simple TagHelper verifier that validates the first few children so these are more easily read.